### PR TITLE
[WIP] Moving files and exclude files and paths

### PR DESCRIPTION
**Pull request by codingHahn**
Needs #4

### CODE REVIEW

File: src/configuration.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// TODO: Make the ConfStruct a global singleton
```
Comment (LucRome) [inline src/configuration.rs:1]: Great idea! I'll create an Issue so we can discuss it.

File: src/main.rs [hunk 2]
```diff
@@ -1,5 +1,29 @@
...
+    let path_wo_excludes: Vec<PathBuf> = path_filter::filter_paths(files_in_source, &config);
+
+    println!("Preliminary file list without excludes: {:?}", path_wo_excludes);
+
+    let dest_folder = std::path::Path::new(&config.destination);
+
+    // Backup every found file
+    for path in path_wo_excludes {
+        backup::backup(&path, &dest_folder).unwrap();
+    }
```
Comment (LucRome) [inline src/main.rs]: Maybe put these in a seperate Function? (Altho I don't know if it's worth the hustle)

File: src/path_filter.rs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    // TODO: Handle when destination is in source
+    let mut result: Vec<PathBuf> = Vec::new();
+    for path in paths {
+        match path {
+            Err(why) => panic!("Something bad happened: {}", why),
+            Ok(path) => {
+                let p = path.file_name().into_string().unwrap();
+                if !conf.exclude_patterns.contains(&p) {
+                    result.push(path.path());
+                }
```
Comment (LucRome) [inline src/path_filter.rs]: Also check the Regex (merge the main Branch into this Branch to to use them), otherwise it seems perfect

**Comment by LucRome** [comment]
Nice, it looks really good and it also seems to be workin

**Comment by codingHahn** [comment]
We need to figure out how to test this function. It is reliant on filesystem state, which is a problem in static unit tests. We could write integration tests in something like python or go the painful way: mocking the rust io API

**Review by LucRome — APPROVED** [review]

**Pull request merged**